### PR TITLE
[feat]: [PL-24921]: Captured audit for ng variable create event

### PR DESCRIPTION
**Pull request by nishantsaini20**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)

### CODE REVIEW

**Review by piyush-harness — CHANGES_REQUESTED** [review]
> Reviewed 3 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)* status: 8 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/events/VariableCreateEvent.java` line 56 at r1](https://reviewable.io/reviews/harness/harness-core/32718#-N17fP-L11VT7Rs-xnrF:-N17fP-M6QXT_5DrwSNL:b-fqsjnr) ([raw file](https://github.com/harness/harness-core/blob/faaa2ef85cd69d4ea59b91debdc2c5c7672c0fc9/120-ng-manager/src/main/java/io/harness/ng/core/events/VariableCreateEvent.java#L56)):*
> ```Java
>   }
> 
>   @JsonIgnore
> ```

Is this intentional?

**Review by nishantsaini20 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)* status: 8 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/events/VariableCreateEvent.java` line 56 at r1](https://reviewable.io/reviews/harness/harness-core/32718#-N17fP-L11VT7Rs-xnrF:-N186q_P5JAGfAqpBrP5:b-2yhjy0) ([raw file](https://github.com/harness/harness-core/blob/faaa2ef85cd69d4ea59b91debdc2c5c7672c0fc9/120-ng-manager/src/main/java/io/harness/ng/core/events/VariableCreateEvent.java#L56)):*
<details><summary><i>Previously, piyush-harness (Piyush Patel) wrote…</i></summary><blockquote>

Is this intentional?
</blockquote></details>

Yes. Since it is a get method, therefore while serialising jackson will create a property named as resource if not annotated with @JsonIgnore

File: 120-ng-manager/src/main/java/io/harness/ng/core/outbox/VariableEventHandler.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  public VariableEventHandler(AuditClientService auditClientService) {
+    this.objectMapper = NG_DEFAULT_OBJECT_MAPPER;
+    this.auditClientService = auditClientService;
+  }
+
+  @Override
+  public boolean handle(OutboxEvent outboxEvent) {
+    try {
+      switch (outboxEvent.getEventType()) {
+        case "VariableCreated":
```
Comment (meenakshi-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/outbox/VariableEventHandler.java:50]: use constants?

**Review by meenakshi-harness — APPROVED** [review]

**Review by nishantsaini20 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)* status: 8 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/outbox/VariableEventHandler.java` line 50 at r1](https://reviewable.io/reviews/harness/harness-core/32718#-N13nMmv66_ho1kebGNR-r1-50:-N1CZq-o92lVnYpuuTEj:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/faaa2ef85cd69d4ea59b91debdc2c5c7672c0fc9/120-ng-manager/src/main/java/io/harness/ng/core/outbox/VariableEventHandler.java#L50)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

use constants?
</blockquote></details>

Done.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 6 of 8 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by piyush-harness — APPROVED** [review]
> Reviewed 4 of 8 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)* status: 1 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32718)* status: :shipit: complete! ✓

**Pull request merged**